diff --git a/src/EventSubscriber/ComponentPluginTranslate.php b/src/EventSubscriber/ComponentPluginTranslate.php
index 34cbf57..f1a1cef 100644
--- a/src/EventSubscriber/ComponentPluginTranslate.php
+++ b/src/EventSubscriber/ComponentPluginTranslate.php
@@ -82,7 +82,11 @@ final class ComponentPluginTranslate implements EventSubscriberInterface {
       $section_storage = $this->routeMatch->getParameter('section_storage');
     }
     else {
-      $section_storage = $this->getSectionStorageForEntity($entity);
+      $view_mode = NULL;
+      if (isset($contexts['view_mode'])) {
+        $view_mode = $contexts['view_mode']->getContextValue();
+      }
+      $section_storage = $this->getSectionStorageForEntity($entity, $view_mode);
     }
 
     if (static::isTranslation($section_storage)) {
